pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -318,13 +324,12 @@ func (facts *PipelineRunFacts) DAGExecutionQueue() (PipelineRunState, error) {
 }
 
 // GetFinalTasks returns a list of final tasks without any taskRun associated with it
-// GetFinalTasks returns final tasks only when all DAG tasks have finished executing successfully or skipped or
-// any one DAG task resulted in failure
```

Comment:
doesn't this still apply? `isDone` implies skipped, succeeded or failed

Reply:
This comment is being updated to reflect the current behavior of GetFinalTasks, since the desired behavior is still under discussion in #4651. GetFinalTasks will not return any tasks if one DAG task resulted in failure but other tasks are still running. Instead, it waits for all tasks to finish before returning.

Reply:
given that we agreed on the behavior #4651, can we update this comment to reflect that? would rather not have the comment reflecting the buggy behavior